### PR TITLE
fix(quantic): improved aria labels of the facet values

### PR DESCRIPTION
**Pull request by mmitiche**
[SVCC-2382](https://coveord.atlassian.net/browse/SVCC-2382)

- Added the facet count ti the aria label of the category facet values.
- Localized the labels used as the aria labels of the facet values.

#### Before: 
we were not using localized labels so we didn't support plural and singular labels.

https://user-images.githubusercontent.com/86681870/221673259-519d2719-6618-4291-8cfe-7aef9a7f31f7.mov

#### After: 
Now we are using localized labels so we do support plural and singular labels.

https://user-images.githubusercontent.com/86681870/221673295-d58b5b91-b15b-4595-b2d0-6729d34b0a11.mov

[SVCC-2382]: https://coveord.atlassian.net/browse/SVCC-2382?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by SimonMilord — APPROVED** [review]

**Review by nathanlb — APPROVED** [review]
Nice!

File: packages/quantic/force-app/main/default/lwc/quanticCategoryFacetValue/quanticCategoryFacetValue.js [hunk 1]
```diff
@@ -64,8 +69,20 @@ export default class QuanticCategoryFacetValue extends LightningElement {
...
   }
 
+  get numberOfResults() {
+    return new Intl.NumberFormat(LOCALE).format(this.item.numberOfResults);
+  }
+
   get ariaLabelValue() {
-    return `Inclusion filter on ${this.facetValue}`;
+    const labelName = I18nUtils.getLabelNameWithCount(
+      'inclusionFilter',
```
Comment (lbergeron) [inline packages/quantic/force-app/main/default/lwc/quanticCategoryFacetValue/quanticCategoryFacetValue.js:78]: I'd suggest to use the same casing as the custom label declaration.

```suggestion
      'InclusionFilter',
```

File: packages/quantic/force-app/main/default/lwc/quanticFacetValue/quanticFacetValue.js [hunk 2]
```diff
@@ -78,7 +86,15 @@ export default class QuanticFacetValue extends LightningElement {
   }
 
   get ariaLabelValue() {
-    return `Inclusion filter on ${this.formattedFacetValue}; ${this.numberOfResults} result`;
+    const labelName = I18nUtils.getLabelNameWithCount(
+      'inclusionFilter',
```
Comment (lbergeron) [inline packages/quantic/force-app/main/default/lwc/quanticFacetValue/quanticFacetValue.js:90]:
```suggestion
      'InclusionFilter',
```

File: packages/quantic/force-app/main/default/labels/CustomLabels.labels-meta.xml [hunk 3]
```diff
@@ -1050,4 +1050,18 @@
       <protected>false</protected>
       <shortDescription>People also ask</shortDescription>
     </labels>
+    <labels>
+      <fullName>quantic_InclusionFilter</fullName>
```
Comment (lbergeron) [inline packages/quantic/force-app/main/default/labels/CustomLabels.labels-meta.xml:1054]: You should also define a custom label suffixed with `_zero`. Even if a facet should not display a value matching no items, it is possible to get into this situation when loading a search page from a bookmark, for instance.

**Review by lbergeron — COMMENTED** [review]
Please add a label suffixed with `_zero`.

File: packages/quantic/force-app/main/default/labels/CustomLabels.labels-meta.xml (hunk 3 above)
Comment (mmitiche) [reply packages/quantic/force-app/main/default/labels/CustomLabels.labels-meta.xml:1054]: good catch! thank you!

**Comment by lvu285** [comment]
can you do a quick check for CategoryFacet, to make sure it works too, thanks!
(your example is for normal facet)

**Comment by mmitiche** [comment]
> can you do a quick check for CategoryFacet, to make sure it works too, thanks!
> (your example is for normal facet)

Yes sure I've already manually tested the category facet as well!

File: packages/quantic/force-app/main/default/lwc/quanticCategoryFacetValue/quanticCategoryFacetValue.js (hunk 1 above)
Comment (mmitiche) [reply packages/quantic/force-app/main/default/lwc/quanticCategoryFacetValue/quanticCategoryFacetValue.js:78]: this key will be used to get the right label from the labels object which is declared as the following:
```
  labels = {
     inLabel,
     inclusionFilter,
     inclusionFilter_plural,
   };
```
That's why I'm using a lower case first letter.

Comment (lbergeron) [reply packages/quantic/force-app/main/default/lwc/quanticCategoryFacetValue/quanticCategoryFacetValue.js:78]: Ah right, my bad :facepalm:

File: packages/quantic/force-app/main/default/lwc/quanticFacetValue/quanticFacetValue.js (hunk 2 above)
Comment (lbergeron) [reply packages/quantic/force-app/main/default/lwc/quanticFacetValue/quanticFacetValue.js:90]: nevermind

File: packages/quantic/force-app/main/default/lwc/quanticCategoryFacetValue/quanticCategoryFacetValue.js [hunk 4]
```diff
@@ -1,6 +1,9 @@
-import {api, LightningElement} from 'lwc';
-
+import LOCALE from '@salesforce/i18n/locale';
 import inLabel from '@salesforce/label/c.quantic_InLabel';
+import inclusionFilter from '@salesforce/label/c.quantic_InclusionFilter';
+import inclusionFilter_plural from '@salesforce/label/c.quantic_InclusionFilter_plural';
```
Comment (lbergeron) [inline packages/quantic/force-app/main/default/lwc/quanticCategoryFacetValue/quanticCategoryFacetValue.js:4]: The `_zero` label should also be included.

File: packages/quantic/force-app/main/default/lwc/quanticFacetValue/quanticFacetValue.js [hunk 5]
```diff
@@ -1,5 +1,8 @@
-import {LightningElement, api} from 'lwc';
 import LOCALE from '@salesforce/i18n/locale';
+import inclusionFilter from '@salesforce/label/c.quantic_InclusionFilter';
+import inclusionFilter_plural from '@salesforce/label/c.quantic_InclusionFilter_plural';
```
Comment (lbergeron) [inline packages/quantic/force-app/main/default/lwc/quanticFacetValue/quanticFacetValue.js:3]: The `_zero` label should also be included.

**Review by lbergeron — CHANGES_REQUESTED** [review]
The `_zero` label has been added but is not used by components.

File: packages/quantic/force-app/main/default/lwc/quanticCategoryFacetValue/quanticCategoryFacetValue.js (hunk 4 above)
Comment (mmitiche) [reply packages/quantic/force-app/main/default/lwc/quanticCategoryFacetValue/quanticCategoryFacetValue.js:4]: oops my bad, just added it, thank you!

**Review by lbergeron — APPROVED** [review]

**Pull request merged**